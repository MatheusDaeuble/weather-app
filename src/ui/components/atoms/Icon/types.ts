import { Color } from 'ui/theme/colors';
import { ViewStyle } from 'react-native';

export type IconName =
  | 'cloud-outline'
  | 'cloudy-night-outline'
  | 'cloudy'
  | 'sync'
  | 'x-circle-outlined'
  | 'check-circle'
  | 'moon-outline'
  | 'partly-sunny-outline'
  | 'rainy-outline'
  | 'snow-outline'
  | 'sunny-outline'
  | 'thunderstorm-outline'
  | 'wind'
  | 'chevron-down'
  | 'chevron-up';

export type IconColor = Color;

export type IconProps = {
  name: IconName;
  disabled?: boolean;
  color?: IconColor;
  size?: number;
  style?: ViewStyle;
  onPress?: () => void;
};
