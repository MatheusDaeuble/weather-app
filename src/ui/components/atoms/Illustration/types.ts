import { SvgProps } from 'react-native-svg';

export type IllustrationName =
  | 'cloud'
  | 'error'
  | 'cloud-moon'
  | 'clouds'
  | 'cloud-sun'
  | 'location'
  | 'mist'
  | 'moon'
  | 'rain'
  | 'rain-moon'
  | 'rain-sun'
  | 'snow'
  | 'storm'
  | 'sun';

export interface IllustrationProps extends SvgProps {
  name: IllustrationName;
}
