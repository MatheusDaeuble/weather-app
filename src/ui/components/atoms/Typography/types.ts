import { TextProps, TextStyle } from 'react-native';
import { Color } from 'ui/theme/colors';

export type TypographyVariant =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'h5'
  | 'body1'
  | 'body2';

export type TypographyProps = {
  variant?: TypographyVariant;
  color?: Color;
  align?: TextStyle['textAlign'];
  children: React.ReactNode;
  paragraph?: boolean;
  onPress?: () => void;
} & TextProps;
