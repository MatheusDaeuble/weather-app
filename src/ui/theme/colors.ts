export enum Colors {
  // PRIMARY
  PRIMARY = '#2E89DF',

  // SECONDARY
  SECONDARY = '#809FFF',

  WHITE = '#FFFFFF',
  BLACK = '#000000',

  // FEEDBACK COLORS
  SUCCESS = '#0E934C',
  WARNING = '#FF8B2C',
  DANGER = '#EB5757',

  // GRAY SCALE
  GRAY_DARK = '#4B4F58',
  GRAY = '#949494',
  GRAY_LIGHT = '#CECECE',
}

export type Color =
  | 'primary'
  | 'secondary'
  | 'white'
  | 'black'
  | 'success'
  | 'warning'
  | 'danger'
  | 'gray-dark'
  | 'gray'
  | 'gray-light';

export const colors: { [key in Color]: string } = {
  'primary': Colors.PRIMARY,
  'secondary': Colors.SECONDARY,
  'white': Colors.WHITE,
  'black': Colors.BLACK,
  'success': Colors.SUCCESS,
  'danger': Colors.DANGER,
  'warning': Colors.WARNING,
  'gray-dark': Colors.GRAY_DARK,
  'gray': Colors.GRAY,
  'gray-light': Colors.GRAY_LIGHT,
};

export const defaultTheme = {
  colors,
};
