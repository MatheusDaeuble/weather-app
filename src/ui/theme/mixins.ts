import { Dimensions, PixelRatio } from 'react-native';

const { width: WINDOW_WIDTH, height: WINDOW_HEIGHT } = Dimensions.get('window');
const guidelineBaseWidth = 375;
const guidelineBaseHeight = 700;

export const scaleSize = (size: number) =>
  Number(((WINDOW_WIDTH / guidelineBaseWidth) * size).toFixed(0));

export const scaleHeight = (size: number) =>
  Number(((WINDOW_HEIGHT / guidelineBaseHeight) * size).toFixed(0));

export const scaleFont = (size: number) =>
  Number((size * PixelRatio.getFontScale()).toFixed(0));
