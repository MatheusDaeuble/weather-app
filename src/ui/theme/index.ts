import * as Colors from './colors';
import * as Spacing from './spacing';
import * as TypographyAttributes from './typography';
import * as Mixins from './mixins';

export { TypographyAttributes, Spacing, Colors, Mixins };
