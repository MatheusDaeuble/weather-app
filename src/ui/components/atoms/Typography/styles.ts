import styled from 'styled-components/native';
import { TypographyAttributes as Typography } from 'ui/theme';
import { colors } from 'ui/theme/colors';
import { TypographyProps } from './types';

const typographyVariant = {
  h1: `
    font-family: ${Typography.FONT_FAMILY_REGULAR};
    font-size: ${Typography.FONT_SIZE_64}px;
    line-height: ${Typography.LINE_HEIGHT_93}px;
  `,
  h2: `
    font-family: ${Typography.FONT_FAMILY_BOLD};
    font-size: ${Typography.FONT_SIZE_26}px;
    line-height: ${Typography.LINE_HEIGHT_39}px;
  `,
  h3: `
    font-family: ${Typography.FONT_FAMILY_BOLD};
    font-size: ${Typography.FONT_SIZE_18}px;
    line-height: ${Typography.LINE_HEIGHT_27}px;
  `,
  h4: `
    font-family: ${Typography.FONT_FAMILY_BOLD};
    font-size: ${Typography.FONT_SIZE_16}px;
    line-height: ${Typography.LINE_HEIGHT_24}px;
  `,
  h5: `
    font-family: ${Typography.FONT_FAMILY_REGULAR};
    font-size: ${Typography.FONT_SIZE_16}px;
    line-height: ${Typography.LINE_HEIGHT_24}px;
  `,
  body1: `
    font-family: ${Typography.FONT_FAMILY_REGULAR};
    font-size: ${Typography.FONT_SIZE_14}px;
    line-height: ${Typography.LINE_HEIGHT_21}px;
  `,
  body2: `
    font-family: ${Typography.FONT_FAMILY_BOLD};
    font-size: ${Typography.FONT_SIZE_14}px;
    line-height: ${Typography.LINE_HEIGHT_21}px;
  `,
};

export const Text = styled.Text<TypographyProps>`
  ${({ variant }) => typographyVariant[variant]}
  color: ${({ color }) => colors[color]};
  text-align: ${({ align }) => align};
  ${({ onPress }) => (onPress ? 'text-decoration-line: underline;' : '')}
`;
