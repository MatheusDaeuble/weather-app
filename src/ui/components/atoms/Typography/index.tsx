import React from 'react';
import { Text } from './styles';
import { TypographyProps } from './types';

const Typography = ({
  variant = 'body1',
  color = 'gray-dark',
  align = 'left',
  paragraph = false,
  children,
  ...props
}: TypographyProps) => {
  if (children === '') return null;

  return (
    <Text
      variant={variant}
      color={color}
      align={align}
      paragraph={paragraph}
      {...props}
    >
      {children}
    </Text>
  );
};

export default Typography;
