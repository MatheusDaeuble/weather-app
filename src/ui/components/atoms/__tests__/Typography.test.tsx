import React from 'react';

import renderer from 'react-test-renderer';
import { Color, colors } from 'ui/theme/colors';
import { TextStyle } from 'react-native';
import { TypographyVariant } from '../Typography/types';
import Typography from '../Typography';

const variants: Array<TypographyVariant> = [
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'body1',
  'body2',
];

const aligns: Array<TextStyle['textAlign']> = [
  'auto',
  'center',
  'justify',
  'left',
  'right',
];

describe('Typography Atom', () => {
  describe('Render variant', () => {
    variants.forEach(variant => {
      it(`Render variant: ${variant}`, () => {
        const component = renderer
          .create(<Typography variant={variant}>{variant}</Typography>)
          .toJSON();
        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('Render color', () => {
    Object.keys(colors).forEach((color: Color) => {
      it(`Render color: ${color}`, () => {
        const component = renderer
          .create(<Typography color={color}>{color}</Typography>)
          .toJSON();
        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('Render align', () => {
    aligns.forEach(align => {
      it(`Render align: ${align}`, () => {
        const component = renderer
          .create(<Typography align={align}>{align}</Typography>)
          .toJSON();
        expect(component).toMatchSnapshot();
      });
    });
  });

  it('Render empty', () => {
    const component = renderer.create(<Typography>{''}</Typography>).toJSON();
    expect(component).toMatchSnapshot();
  });
});
