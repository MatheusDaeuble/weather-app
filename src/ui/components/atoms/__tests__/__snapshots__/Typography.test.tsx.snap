// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Atom Render align Render align: auto 1`] = `
<Text
  align="auto"
  color="gray-dark"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#4B4F58",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "auto",
      },
    ]
  }
  variant="body1"
>
  auto
</Text>
`;

exports[`Typography Atom Render align Render align: center 1`] = `
<Text
  align="center"
  color="gray-dark"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#4B4F58",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "center",
      },
    ]
  }
  variant="body1"
>
  center
</Text>
`;

exports[`Typography Atom Render align Render align: justify 1`] = `
<Text
  align="justify"
  color="gray-dark"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#4B4F58",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "justify",
      },
    ]
  }
  variant="body1"
>
  justify
</Text>
`;

exports[`Typography Atom Render align Render align: left 1`] = `
<Text
  align="left"
  color="gray-dark"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#4B4F58",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "left",
      },
    ]
  }
  variant="body1"
>
  left
</Text>
`;

exports[`Typography Atom Render align Render align: right 1`] = `
<Text
  align="right"
  color="gray-dark"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#4B4F58",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "right",
      },
    ]
  }
  variant="body1"
>
  right
</Text>
`;

exports[`Typography Atom Render color Render color: black 1`] = `
<Text
  align="left"
  color="black"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "left",
      },
    ]
  }
  variant="body1"
>
  black
</Text>
`;

exports[`Typography Atom Render color Render color: danger 1`] = `
<Text
  align="left"
  color="danger"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#EB5757",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "left",
      },
    ]
  }
  variant="body1"
>
  danger
</Text>
`;

exports[`Typography Atom Render color Render color: gray 1`] = `
<Text
  align="left"
  color="gray"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#949494",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "left",
      },
    ]
  }
  variant="body1"
>
  gray
</Text>
`;

exports[`Typography Atom Render color Render color: gray-dark 1`] = `
<Text
  align="left"
  color="gray-dark"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#4B4F58",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "left",
      },
    ]
  }
  variant="body1"
>
  gray-dark
</Text>
`;

exports[`Typography Atom Render color Render color: gray-light 1`] = `
<Text
  align="left"
  color="gray-light"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#CECECE",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "left",
      },
    ]
  }
  variant="body1"
>
  gray-light
</Text>
`;

exports[`Typography Atom Render color Render color: primary 1`] = `
<Text
  align="left"
  color="primary"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#2E89DF",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "left",
      },
    ]
  }
  variant="body1"
>
  primary
</Text>
`;

exports[`Typography Atom Render color Render color: secondary 1`] = `
<Text
  align="left"
  color="secondary"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#809FFF",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "left",
      },
    ]
  }
  variant="body1"
>
  secondary
</Text>
`;

exports[`Typography Atom Render color Render color: success 1`] = `
<Text
  align="left"
  color="success"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#0E934C",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "left",
      },
    ]
  }
  variant="body1"
>
  success
</Text>
`;

exports[`Typography Atom Render color Render color: warning 1`] = `
<Text
  align="left"
  color="warning"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#FF8B2C",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "left",
      },
    ]
  }
  variant="body1"
>
  warning
</Text>
`;

exports[`Typography Atom Render color Render color: white 1`] = `
<Text
  align="left"
  color="white"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#FFFFFF",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "left",
      },
    ]
  }
  variant="body1"
>
  white
</Text>
`;

exports[`Typography Atom Render empty 1`] = `null`;

exports[`Typography Atom Render variant Render variant: body1 1`] = `
<Text
  align="left"
  color="gray-dark"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#4B4F58",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "left",
      },
    ]
  }
  variant="body1"
>
  body1
</Text>
`;

exports[`Typography Atom Render variant Render variant: body2 1`] = `
<Text
  align="left"
  color="gray-dark"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#4B4F58",
        "fontFamily": "OpenSans_700Bold",
        "fontSize": 28,
        "lineHeight": 42,
        "textAlign": "left",
      },
    ]
  }
  variant="body2"
>
  body2
</Text>
`;

exports[`Typography Atom Render variant Render variant: h1 1`] = `
<Text
  align="left"
  color="gray-dark"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#4B4F58",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 124,
        "lineHeight": 186,
        "textAlign": "left",
      },
    ]
  }
  variant="h1"
>
  h1
</Text>
`;

exports[`Typography Atom Render variant Render variant: h2 1`] = `
<Text
  align="left"
  color="gray-dark"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#4B4F58",
        "fontFamily": "OpenSans_700Bold",
        "fontSize": 52,
        "lineHeight": 78,
        "textAlign": "left",
      },
    ]
  }
  variant="h2"
>
  h2
</Text>
`;

exports[`Typography Atom Render variant Render variant: h3 1`] = `
<Text
  align="left"
  color="gray-dark"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#4B4F58",
        "fontFamily": "OpenSans_700Bold",
        "fontSize": 36,
        "lineHeight": 54,
        "textAlign": "left",
      },
    ]
  }
  variant="h3"
>
  h3
</Text>
`;

exports[`Typography Atom Render variant Render variant: h4 1`] = `
<Text
  align="left"
  color="gray-dark"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#4B4F58",
        "fontFamily": "OpenSans_700Bold",
        "fontSize": 32,
        "lineHeight": 48,
        "textAlign": "left",
      },
    ]
  }
  variant="h4"
>
  h4
</Text>
`;

exports[`Typography Atom Render variant Render variant: h5 1`] = `
<Text
  align="left"
  color="gray-dark"
  paragraph={false}
  style={
    Array [
      Object {
        "color": "#4B4F58",
        "fontFamily": "OpenSans_400Regular",
        "fontSize": 32,
        "lineHeight": 48,
        "textAlign": "left",
      },
    ]
  }
  variant="h5"
>
  h5
</Text>
`;
